File: samples-and-tests/yabe/app/models/Tag.java
package models;
 
import java.util.*;
import javax.persistence.*;
 
import play.db.jpa.*;
import play.data.validation.*;
 
@Entity
public class Tag extends Model implements Comparable<Tag> {
 
    @Required
    public String name;
    
    private Tag(String name) {
        this.name = name;
    }
    
    public static Tag findOrCreateByName(String name) {
        Tag tag = Tag.find("byName", name).first();
        if(tag == null) {
            tag = new Tag(name);
        }
        return tag;
    }
    
    public static List<Map> getCloud() {
        List<Map> result = Tag.find(
            "select new map(t.name as tag, count(p.id) as pound) from Post p join p.tags as t group by t.name"
        ).fetch();
        return result;
    }
    
    public String toString() {
        return name;
    }
    
    public int compareTo(Tag otherTag) {
        return name.compareTo(otherTag.name);
    }
 
}

File: samples-and-tests/yabe-with-scala/app/models/Tag.scala
package models
 
import java.util._
import javax.persistence._
 
import play.db.jpa._
import play.data.validation._
 
@Entity
class Tag private (

    @Required 
    var name:String
    
) extends Model[Tag] with Comparable[Tag] {
    
    override def toString() = {
        name
    }
    
    override def compareTo(otherTag: Tag) = {
        name.compareTo(otherTag.name)
    }
 
}

object Tag extends Model[Tag]{
    
    def allTags = Tag.findAll
    
    def findOrCreateByName(name: String) = {
        var tag = Tag.find("byName", name).first
        if(tag == null) {
            tag = new Tag(name)
        }
        tag
    }
    
    def cloud = {
        Tag.find(
            "select new map(t.name as tag, count(p.id) as pound) from Post p join p.tags as t group by t.name"
        ).fetch
    }
    
}
